### PR TITLE
fix/5812

### LINKED ISSUE #5812: When using data-search-selector it doesn't detect search-text
**Bootstraptable version(s) affected**:
All? tried on 5 and on the live editor.

**Description**  
I am setting up a data-search-selector to use sticky search and search-text just change the value of that input, but doesn't execute the search.

**Example**  
https://live.bootstrap-table.com/code/eliezra236/8439

**Possible  (optional)**  

I made some attempts of changing it way later (trough button that execute:   
```
$("#table").bootstrapTable("refreshOptions", {
    searchText: "testing",
  });
```
And it does the same thing, changes the text in the input but doesn't execute the search.

### PR DESCRIPTION
**Pull request by UtechtDustin**
**🤔Type of Request**
- [x] **Bug fix**
- [ ] **New feature**
- [ ] **Improvement**
- [ ] **Documentation**
- [ ] **Other**

**🔗Resolves an issue?**

Fix #5812
**📝Changelog**

- [x] **Core**
- [ ] **Extensions**

Fixed the comparision of search inputs (DOM element and jQuery Element/Object).

**💡Example(s)?**

Before: https://live.bootstrap-table.com/code/eliezra236/8439
After: https://live.bootstrap-table.com/code/UtechtDustin/8513

**☑️Self Check before Merge**

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -913,7 +913,11 @@ class BootstrapTable {
         return
       }
 
-      if (currentTarget === Utils.getSearchInput(this)[0] || $(currentTarget).hasClass('search-input')) {
+      if (
+        (currentTarget instanceof jQuery && currentTarget.is(Utils.getSearchInput(this))) ||
```
Comment (djhvscf) [inline src/bootstrap-table.js:917]: What about to get the `Utils.getSearchInput(this)` in a variable and then just evaluate it in this if statement?

File: src/bootstrap-table.js [hunk 2]
```diff
@@ -913,7 +913,11 @@ class BootstrapTable {
...
+        (!(currentTarget instanceof jQuery) && currentTarget === Utils.getSearchInput(this)[0]) ||
+        $(currentTarget).hasClass('search-input')
```
Comment (djhvscf) [inline src/bootstrap-table.js:919]: At the end we could convert currentTarget to a jQuery object and then evaluate if that object has the corresponding class.. I don't think we could have performance hit if we create a jQuery object from a jQuery object... Do you know that?

**Review by djhvscf — APPROVED** [review]

**Review by djhvscf — CHANGES_REQUESTED** [review]
please review my previous comments

File: src/bootstrap-table.js (hunk 2 above)
Comment (UtechtDustin) [reply src/bootstrap-table.js:919]: I'm not sure what do you mean, we don't create a jQuery object on that if.
We just check if the currentTarget is a jQuery object, if it is a jQuery we compare it using the `is` function.
If it is a dom element we get the dom element from the jquery object and just compare it with `===`

File: src/bootstrap-table.js (hunk 1 above)
Comment (UtechtDustin) [reply src/bootstrap-table.js]: I agree it's a micro optimization :-)

File: src/bootstrap-table.js (hunk 2 above)
Comment (djhvscf) [reply src/bootstrap-table.js:921]: What about this?
```js
const searchInput = Utils.getSearchInput(this)
currentTarget = $(currentTarget)
if (
        currentTarget.is(searchInput) ||
        currentTarget.hasClass('search-input')
...
```

Comment (UtechtDustin) [reply src/bootstrap-table.js:921]: I guess creating a new jquery object will cost more performance.
So i could invert it, i can check if it is a jQuery Object and set `currentTarget` to a domElement.
In the if i would use the vanilla js functions for the comparison.

What do you mean ? :-)

**Comment by UtechtDustin** [comment]
@djhvscf please check again.

**Review by djhvscf — APPROVED** [review]

**Pull request merged**